### ISSUE TITLE
Create an observable version of the block tree

### ISSUE THREAD

**Issue by kevinputera** [AY1920S2-CS2103T-W17-2/main#81]
To enable observability in our `BlockManager` model (see #8), we need a custom observable implementation of the `Block` tree. That is, this custom class should support all the CRUD operations of the `Block` tree, e.g. create a `Block` at a particular location, and propagate the changes to all the listeners attached to it.

**Comment by kevinputera** [comment]
Proposed API

```java
public interface ObservableBlockTree {
  Block getRootBlock();
  void add(Path path, Block block);
  void set(Path path, Block block);
  void remove(Path path);
}
```

**Comment by kevinputera** [comment]
@firzanarmani Hey firzan, please take a look and let me know what you think!

**Comment by firzanarmani** [comment]
@kevinputera I'll try to implement this and see if I encounter any problems. Will keep you updated!

**Comment by kevinputera** [comment]
@firzanarmani FYI, I was mirroring the method namings of the `List` class. That's why we have the `add`, `set`, and the `remove` methods!

**Comment by firzanarmani** [comment]
Closed by #151
